frontend/src/metabase/visualizations/visualizations/PieChart/chart-definition.ts

```diff
@@ -119,16 +127,21 @@ export const PIE_CHART_DEFINITION: VisualizationDefinition = {
             String(formatValue(value, options)),
           );
         },
-        (rawSeries, settings) =>
-          JSON.stringify(rawSeries) + JSON.stringify(settings),
+        ([{ card, data }], settings) =>
+          JSON.stringify({
+            cols: data.cols,
+            rows: data.rows.length,
+            dataset_query: card.dataset_query,
```

Comment:
I think cols and rows length can remain the same if underlying data has 1 row modified. 
The `/api/dataset` endpoint returns `started_at` what should make it unique enough. What do you think if we combine it with the `dataset_query`?

Reply:
@alxnddr good call, I had to update the way we're building a series object in a few places to make sure they include `started_at` apart from `card` and `data`. Please take another look